Docker 1.4.1 panics when starting container in docker/volumes/volume.go:78

Hi,

just rebooted a Docker 1.4.1 host. On start it tried to start old containers which caused it to panic:

```
ESC[34mDEBUESC[0m[0004] Starting container 648be33493a1c2fcfa1b305bb69d5b7c8ae4fa4a6338e5b6634d26b9494a40e8 
ESC[34mINFOESC[0m[0004] +job allocate_interface(648be33493a1c2fcfa1b305bb69d5b7c8ae4fa4a6338e5b6634d26b9494a40e8) 
ESC[34mINFOESC[0m[0004] -job allocate_interface(648be33493a1c2fcfa1b305bb69d5b7c8ae4fa4a6338e5b6634d26b9494a40e8) = OK (0) 
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x6c8024]

goroutine 24 [running]:
runtime.panic(0xbe5c40, 0x127da13)
        /usr/local/go/src/pkg/runtime/panic.c:279 +0xf5
github.com/docker/docker/volumes.(*Volume).RemoveContainer(0x0, 0xc20871a780, 0x40)
        /go/src/github.com/docker/docker/volumes/volume.go:78 +0x94
github.com/docker/docker/daemon.(*Mount).initialize(0xc208bfeea0, 0x0, 0x0)
        /go/src/github.com/docker/docker/daemon/volumes.go:93 +0x1b3
github.com/docker/docker/daemon.(*Container).createVolumes(0xc2085ea340, 0x0, 0x0)
        /go/src/github.com/docker/docker/daemon/volumes.go:69 +0xd6
github.com/docker/docker/daemon.(*Container).prepareVolumes(0xc2085ea340, 0x0, 0x0)
        /go/src/github.com/docker/docker/daemon/volumes.go:48 +0x5e
github.com/docker/docker/daemon.(*Container).Start(0xc2085ea340, 0x0, 0x0)
        /go/src/github.com/docker/docker/daemon/container.go:343 +0x28c
github.com/docker/docker/daemon.(*Daemon).restore(0xc2080d20b0, 0x0, 0x0)
        /go/src/github.com/docker/docker/daemon/daemon.go:386 +0xf3a
github.com/docker/docker/daemon.NewDaemonFromDirectory(0x12a0700, 0xc2080d5340, 0x0, 0x0, 0x0)
        /go/src/github.com/docker/docker/daemon/daemon.go:924 +0x1ffa
github.com/docker/docker/daemon.NewDaemon(0x12a0700, 0xc2080d5340, 0x0, 0x0, 0x0)
        /go/src/github.com/docker/docker/daemon/daemon.go:719 +0x43
main.func·001()
        /go/src/github.com/docker/docker/docker/daemon.go:53 +0x46
created by main.mainDaemon
        /go/src/github.com/docker/docker/docker/daemon.go:76 +0x2e4

goroutine 16 [chan receive]:
github.com/docker/docker/api/server.ServeApi(0xc20804cc00, 0x7f30ed6417d8)
        /go/src/github.com/docker/docker/api/server/server.go:1573 +0x473
github.com/docker/docker/engine.(*Job).Run(0xc20804cc00, 0x0, 0x0)
        /go/src/github.com/docker/docker/engine/job.go:83 +0x837
main.mainDaemon()
        /go/src/github.com/docker/docker/docker/daemon.go:91 +0x568
main.main()
        /go/src/github.com/docker/docker/docker/docker.go:71 +0x3ff

...
```

This might or might not be related to this "creative" bind-mount:
```
   "Volumes" : {
      "/var/run/docker.sock" : "/run/docker.sock",
      "/host/proc/mounts" : "/proc/10453/mounts",
      "/host/sys/fs/cgroup" : "/sys/fs/cgroup"
   },
```

Which doesn't look right but shouldn't cause Docker to crash either.

--- a/daemon/volumes.go
+++ b/daemon/volumes.go
@@ -89,9 +89,10 @@ func (m *Mount) initialize() error {
 
 		// Make sure we remove these old volumes we don't actually want now.
 		// Ignore any errors here since this is just cleanup, maybe someone volumes-from'd this volume
-		v := m.container.daemon.volumes.Get(hostPath)
-		v.RemoveContainer(m.container.ID)
-		m.container.daemon.volumes.Delete(v.Path)
+		if v := m.container.daemon.volumes.Get(hostPath); v != nil {
+			v.RemoveContainer(m.container.ID)
+			m.container.daemon.volumes.Delete(v.Path)
+		}
 	}
 
 	// This is the full path to container fs + mntToPath

